patterns/direct_processor.py

```diff
@@ -0,0 +1,142 @@
+        print(f"Message: {e.get_message()}")
+
+    def on_service_interrupted(self, e: "ServiceEvent"):
+        print("\non_service_interrupted")
+        print(f"Error cause: {e.get_cause()}")
+        print(f"Message: {e.get_message()}")
+
+# Event Handling for the messaging service
+service_handler = ServiceEventHandler()
+messaging_service.add_reconnection_listener(service_handler)
+messaging_service.add_reconnection_attempt_listener(service_handler)
+messaging_service.add_service_interruption_listener(service_handler)
+ 
+# Inner classes for publish error handling
+class PublisherErrorHandling(PublishFailureListener):
+    def on_failed_publish(self, e: "FailedPublishEvent"):
+        print("on_failed_publish")
+
+# Handle received messages
+class MessageHandlerImpl(MessageHandler):
+    def on_message(self, message: InboundMessage):
+        out_topic = Topic.of(TOPIC_PREFIX + f'/direct/processor/output')
+        in_topic = message.get_destination_name()
+        payload_str = message.get_payload_as_string()
+        print(f'Received input message (body string):\n{payload_str}')
+        print(f'Message dump: {message.solace_message.get_message_dump()} \n')
+
+        processed_payload_str = payload_str.upper()
+        print(f'Published output message (body):\n{processed_payload_str}')
+
+        output_msg = msg_builder \
+                        .with_application_message_id(message.get_application_message_id())\
+                        .build(f'{processed_payload_str}')
+
+        direct_publisher.publish(destination=out_topic, message=output_msg)
+
+        print(f'<<<<<<<<<<<<<<<<<<<<<<<<<<<<')
+        print(f'Received input message on {in_topic}')
+        print(f'Received input message (body):\n{payload_str}')
+        print(f'----------------------------')
+        print(f'>>>>>>>>>>>>>>>>>>>>>>>>>>>>')
+        print(f'Published output message on {out_topic}')
+        print(f'Published output message (body):\n{processed_payload_str}')
+        print(f'----------------------------')
+
+# Create a direct message publisher and start it
+direct_publisher = messaging_service.create_direct_message_publisher_builder().build()
+direct_publisher.set_publish_failure_listener(PublisherErrorHandling())
+
+# Blocking Start thread
+direct_publisher.start()
+print(f'Direct Publisher ready? {direct_publisher.is_ready()}')
+
+# Define a Topic subscriptions 
+topics = [TOPIC_PREFIX + "/direct/processor/input/>"]
+topics_sub = []
+for t in topics:
+    topics_sub.append(TopicSubscription.of(t))
+print(f'\nSubscribed to topic {topics}')
+
+# Build a Receiver with the given topics and start it
+direct_receiver = messaging_service.create_direct_message_receiver_builder()\
+                        .with_subscriptions(topics_sub)\
+                        .build()
+direct_receiver.start()
+print(f'Direct Subscriber is running? {direct_receiver.is_running()}')
+
+# Prepare outbound message payload and body
```

Comment:
This block is not needed, `msg_builder` is not used

Reply:
I realized you are using it in the message handler. I suggest moving there

Reply:
Done